vision_ocv/benchmark/modules/src/BM_ocv_vision_factory_test.cpp

```diff
@@ -0,0 +1,168 @@
+// Copyright 2023 Klepsydra Technologies AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "config.h"
+#include <benchmark/benchmark.h>
+#include <klepsydra/core/event_emitter_middleware_provider.h>
+#include <klepsydra/high_performance/event_loop_middleware_provider.h>
+#include <klepsydra/vision_ocv/file_image_stream_service.h>
+#include <klepsydra/vision_ocv/image_data_factory.h>
+#include <opencv2/opencv.hpp>
+#include <spdlog/spdlog.h>
+class FactoryPerformanceTest : public benchmark::Fixture
+{
+public:
+    FactoryPerformanceTest()
+        : benchmark::Fixture()
+        , width(1000)
+        , height(1000)
+        , imageDataFactory(width, height, cv::IMREAD_REDUCED_GRAYSCALE_2, "raw_image")
+        , _cloneF([](const kpsr::vision_ocv::ImageData &src, kpsr::vision_ocv::ImageData &dst) {
+            dst.clone(src);
+        })
+        , _buildF([](kpsr::vision_ocv::ImageData &src) { src = kpsr::vision_ocv::ImageData(src); })
+    {}
+    ~FactoryPerformanceTest() {}
+
+    void SetUp(const ::benchmark::State &state)
+    {
+        bool withFactory = state.range(0);
```

Comment:
This flag is redundant and makes the benchmark to run redundantly.
I'd replace it with: `bool withFactory = state.range(0) == 0;`
(obviously, once you remove this param from the benchmark, all range indexes are minus one):

```
int poolSize = state.range(0);
bool useEventLoop = state.range(1);
```